### ISSUE TITLE
^:figwheel-always in not picked up if the ns is cached

### ISSUE THREAD

**Issue by Deraen** [bhauman/lein-figwheel#124]
If cljs build cache exists `^:figwheel-always` doesn't work until I change the ns. It works fine after clearing the `:output-dir`.

**Comment by bhauman** [comment]
Can you tell me what you are referring to when you say "cljs build cache"? Also what version of ClojureScript are you using?

**Comment by Deraen** [comment]
I mean any files left to :output-dir by cljs compiler. Looks like this is a problem with `ana-api/all-ns`: https://github.com/Deraen/problems/tree/all-ns#log

I'm using cljs 3126 and I verified that the all-ns works the same with 3196.

**Comment by Deraen** [comment]
```
<Deraen> dnolen: Looks like cljs.analyzer.api/all-ns returns nil when starting with existing output-dir and unchanged sources: https://github.com/Deraen/problems/tree/all-ns#log this is a bug, I think?
<dnolen> Deraen: it's debatable really, build doesn't make any promises about populating the analysis environment
<dnolen> Deraen: you probably want to call file-seq on the dir and map analyze-file over that, then call all-ns
```

**Comment by bhauman** [comment]
Deraen thanks for pointing this out.

I'd really like to know more about what you are experiencing more specifically.  Can you give me some quick reproduction steps, a gist or a small project? Any help you can give will be appreciated.

I'm assuming that you are using :cache-analysis true

Actually this is something I think I noticed with the latest cljs.  Some strange things.  Like the initial build doesn't seem to populate the env, presumably because there are no changes, but subsequent ones do because a file has changed, where before the first build would populate the env, even if there are no changes. 

Thanks again for pointing this out.

**Comment by Deraen** [comment]
Ah, I was so focused on finding the cause that I forgot to describe the problem clearly :)

I created a example project here: https://github.com/Deraen/problems/tree/figwheel-reload-always
After starting Figwheel on a clear directory changing any cljs file will trigger core.js reload (as the core namespace has the figwheel-always metadata). 
But when I run Figwheel again on the same directory Figwheel will not reload the core namespace until the core.cljs file is changed once.

I did try with and without :cache-analysis and it doesn't seem to affect this.
